app/assets/javascripts/discourse/app/components/sidebar/user/custom-sections.js

```diff
@@ -39,10 +39,14 @@ export default class SidebarUserCustomSections extends Component {
           ? htmlSafe(`${iconHTML("globe")} ${section.title}`)
           : section.title;
       section.links.forEach((link) => {
-        const routeInfoHelper = new RouteInfoHelper(this.router, link.value);
-        link.route = routeInfoHelper.route;
-        link.models = routeInfoHelper.models;
-        link.query = routeInfoHelper.query;
+        if (link.value.startsWith("http", "https")) {
+          link.external = true;
```

Comment:
This is slightly confusing for me because a link is only considered external if the host does not match. It shouldn't just be based on whether a protocol is present?

Reply:
It is valid, but maybe we can make it easier to follow.

The reason is that internal links are stored as paths. So even if type https://meta.discourse.org/tags, only /tags is stored in database. The reason is that when you change hostname of your site, internal links will still work.

External links are kept in original form so whole `https://discourse.org` is stored in database.

That's why, it is enough that when URL has http it means it is external

Reply:
Ahh icic. This is when the link has already been stored in the database. I thought of this as adding a new link.